src/main/java/org/folio/inventory/validation/InstanceSourceTypeValidator.java

```diff
@@ -0,0 +1,34 @@
+package org.folio.inventory.validation;
+
+import org.folio.inventory.domain.instances.Instance;
+import org.folio.inventory.exceptions.UnprocessableEntityException;
+import org.folio.inventory.support.http.server.ValidationError;
+
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.logging.log4j.util.Strings.isNotBlank;
+import static org.folio.inventory.support.CompletableFutures.failedFuture;
+
+public final class InstanceSourceTypeValidator {
+
+  private static final String[] INSTANCE_SOURCE_TYPES = {"MARC", "FOLIO", "CONSORTIUM-MARC", "CONSORTIUM-FOLIO"};
+
+  private InstanceSourceTypeValidator() {}
+
+  public static CompletableFuture<Instance> refuseWhenIncorrectSource(Instance instance) {
+
+    final ValidationError incorrectSourceError = new ValidationError(
```

Comment:
it seems we could inline the creation of `ValidationError` to line#27, thus it won't be created in case of the correct source type